rust/worker/src/segment/metadata_segment.rs

```diff
@@ -1164,10 +1164,11 @@ impl MetadataSegmentReader<'_> {
         })
     }
 
+    // DEPRECATED: This exists only for the legacy testing. Please checkout `MetadataFilteringOperator` for the up to date implementation.
     pub async fn query(
```

Comment:
curious, why did we not remove it? Is it just more work or are there other considerations?

Reply:
See above